src/utility.rs

```diff
@@ -154,43 +154,23 @@ impl PartialEq<(&str, &str, &str)> for SplittedString<'_> {
 }
 
 /// Convert preceding and trailing quotation marks(', ") into their curved form(‘, ’, “, ”) aka Smart Quote.
-pub(crate) fn smart_quoter(mut splitted: SplittedString) -> SplittedString {
+pub(crate) fn smart_quoter(splitted: SplittedString) -> SplittedString {
     // If the middle part is empty, there is no need to convert.
     if splitted.word().is_empty() {
         return splitted;
     }
 
     // Convert preceding quotation mark(', ") into its curved form(‘, “).
-    let mut preceding = String::with_capacity(splitted.preceding().len() + 3);
-    for ch in splitted.preceding().chars() {
-        match ch {
-            '\'' => {
-                preceding.push_str("‘");
-            }
-            '"' => {
-                preceding.push_str("“");
-            }
-            _ => preceding.push(ch),
-        }
-    }
+    let preceding = splitted.preceding.replace('\'', "‘").replace('"', "“");
```

Comment:
I am quite skeptical about the performance implication of these two changes. It is a hot path, as it'll be always checking the string every time a new suggestion is to be made. It uses the [`Pattern`](https://doc.rust-lang.org/std/str/pattern/trait.Pattern.html) API which seems to  have a lot of indirection. I'd be okay we can make sure the compiler is able to optimize all of these indirection and do not cause any overhead.

How about a benchmark?

Reply:
Yes, there can be a benchmark. I'll once I get some time.

Reply:
@mominul In my simple benchmark, the new code is taking almost double time. Most probably because of double pass and not declaring string with expected higher capacity. Going to close this pr.

Reply:
@gulshan Thanks for checking it! ❤️